src/mode/modeHandler.ts

```diff
@@ -1519,6 +1603,101 @@ export class ModeHandler implements vscode.Disposable {
 
     this.vimState.editor.setDecorations(decoration.Default, cursorRange);
 
+    // Insert Mode virtual characters: used to temporarily show the remapping pressed keys on
+    // insert mode, to show the `"` character after pressing `<C-r>`, and to show `?` and the
+    // first character when inserting digraphs with `<C-k>`.
+    let decorationOptions: vscode.DecorationOptions[] = [];
+    if (
+      (this.vimState.currentMode === Mode.Insert || this.vimState.currentMode === Mode.Replace) &&
+      (vimState.recordedState.bufferedKeys.length > 0 ||
```

Comment:
Remove this `&&` clause and make the `else` on line 1619 `else if (vimState.recordedState.waitingForAnotherActionKey)`